node/pkg/common/chainlock_test.go

```diff
@@ -0,0 +1,105 @@
+package common
+
+import (
+	"github.com/certusone/wormhole/node/pkg/vaa"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func getTestMP() MessagePublication {
+	txHash := common.HexToHash("80d6a81b73b3cebba81fba2a330bcfaa25ce93e52e6bd65a02c7c13932a8c1a5")
+	timeStamp := time.Now()
+	nonce := uint32(1)
+	sequence := uint64(1)
+	consistencyLevel := uint8(1)
+	emitterChain := vaa.ChainIDEthereum
+	emitterAddress := vaa.Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+	payload := []byte{97, 97, 97, 97, 97, 97}
+
+	return MessagePublication{
+		TxHash:           txHash,
+		Timestamp:        timeStamp,
+		Nonce:            nonce,
+		Sequence:         sequence,
+		ConsistencyLevel: consistencyLevel,
+		EmitterChain:     emitterChain,
+		EmitterAddress:   emitterAddress,
+		Payload:          payload,
+	}
+}
+
+// Base test to ensure the
+func TestMessagePublication(t *testing.T) {
+	txHash := common.HexToHash("80d6a81b73b3cebba81fba2a330bcfaa25ce93e52e6bd65a02c7c13932a8c1a5")
+	timeStamp := time.Now()
+	nonce := uint32(1)
+	sequence := uint64(1)
+	consistencyLevel := uint8(1)
+	emitterChain := vaa.ChainIDEthereum
+	emitterAddress := vaa.Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+	payload := []byte{97, 97, 97, 97, 97, 97}
+
+	mp := MessagePublication{
+		TxHash:           txHash,
+		Timestamp:        timeStamp,
+		Nonce:            nonce,
+		Sequence:         sequence,
+		ConsistencyLevel: consistencyLevel,
+		EmitterChain:     emitterChain,
+		EmitterAddress:   emitterAddress,
+		Payload:          payload,
+	}
+
+	assert.Equal(t, txHash, mp.TxHash)
+	assert.Equal(t, timeStamp, mp.Timestamp)
+	assert.Equal(t, nonce, mp.Nonce)
+	assert.Equal(t, sequence, mp.Sequence)
+	assert.Equal(t, consistencyLevel, mp.ConsistencyLevel)
+	assert.Equal(t, emitterChain, mp.EmitterChain)
+	assert.Equal(t, emitterAddress, mp.EmitterAddress)
+	assert.Equal(t, payload, mp.Payload)
+}
```

Comment:
I don't actually see the point of what this test is testing? It is supposed to test that assigning a struct works?

Reply:
It locks in a firm expectation of what the struct will do.  For example, if the struct changes (or behaves differently) it will fail and require the person to revisit the test.   I feel like we should go back vaa/common structs similarly, just to lock in their expected behavior, so they don't change in some unplanned/emergent way.

I think another/additional way this could be spec'd in could be to marshal the struct to a hex string, such that if the struct is iterated on it will require a unit-test change.

Reply:
If someone adds something to the struct, this test will not catch it. You're testing that go works as expected here more than anything else. This is not the kind of thing to make assertions against IMO. If someone changes an existing thing, it will catch it, but this literally is not actually making any assertions other than go works as designed.

I'm not convinced this test has any utility, but I'm still going to approve it.

Reply:
Removed.